src/v0/destinations/sfmc/transform.js

```diff
@@ -34,51 +32,38 @@ const CONTACT_KEY_KEY = 'Contact Key';
 // DOC: https://developer.salesforce.com/docs/atlas.en-us.mc-app-development.meta/mc-app-development/access-token-s2s.htm
 
 const getToken = async (clientId, clientSecret, subdomain) => {
-  try {
-    const resp = await myAxios.post(
-      `https://${subdomain}.${ENDPOINTS.GET_TOKEN}`,
-      {
-        grant_type: 'client_credentials',
-        client_id: clientId,
-        client_secret: clientSecret,
-      },
-      {
-        'Content-Type': JSON_MIME_TYPE,
-      },
-      {
-        destType: 'sfmc',
-        feature: 'transformation',
-        endpointPath: '/token',
-        requestMethod: 'POST',
-        module: 'router',
-      },
-    );
-    if (resp && resp.data) {
-      return resp.data.access_token;
-    }
-    const status = resp.status || 400;
+  const { processedResponse: processedResponseSfmc } = await handleHttpRequest(
+    'post',
+    `https://${subdomain}.${ENDPOINTS.GET_TOKEN}`,
+    {
+      grant_type: 'client_credentials',
+      client_id: clientId,
+      client_secret: clientSecret,
+    },
+    {
+      'Content-Type': JSON_MIME_TYPE,
+    },
+    {
+      destType: 'sfmc',
+      feature: 'transformation',
+      endpointPath: '/token',
+      requestMethod: 'POST',
+      module: 'router',
+    },
+  );
+
+  if (!isHttpStatusSuccess(processedResponseSfmc.status)) {
     throw new NetworkError(
       'Could not retrieve access token',
-      status,
+      processedResponseSfmc.status || 400,
       {
-        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status),
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(processedResponseSfmc.status || 400),
       },
-      resp,
+      JSON.stringify(processedResponseSfmc),
```

Comment:
Should we stringify ?

Reply:
in consistent ? I didn't get it

Reply:
other destinations return `processedResponseSfmc response` instead of `processedResponseSfmc` in NetworkError

Reply:
When you're setting the last argument, you're also setting `processedResopnseSfmc.response` itself right ? It is consistent with other destinations correct ?

Reply:
yes

Reply:
wdyt?

Reply:
Yeah. May be we can send `processedResponseSfmc` as if we only send `processedResponseSfmc.response` you're just sending response body and response status-code is not present in `destinationResponse`.